src/SprykerMiddleware/Zed/Process/Business/PluginFinder/PluginFinder.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace SprykerMiddleware\Zed\Process\Business\PluginFinder;
+
+use SprykerMiddleware\Zed\Process\Business\Exception\IteratorNotConfiguredException;
+use SprykerMiddleware\Zed\Process\Business\Exception\StagePluginsNotConfiguredException;
+use SprykerMiddleware\Zed\Process\Dependency\Plugin\Iterator\ProcessIteratorPluginInterface;
+use SprykerMiddleware\Zed\Process\ProcessConfig;
+
+class PluginFinder implements PluginFinderInterface
```

Comment:
Violation of Single resp and Interface segr principles.
Does it make sense to split this one finder to a couple of more specific: StageFinder, ProcessorHookFinder etc.